src/components/CowPenContextMenu/CowPenContextMenu.test.js

```diff
@@ -125,90 +161,296 @@ describe('CowCard', () => {
+        baseWeight: 100,
+        gender: genders.MALE,
+      })
+      const breedingCow2 = generateCow({
+        color: cowColors.WHITE,
+        name: '',
+        baseWeight: 100,
+        gender: genders.FEMALE,
+      })
+
+      render(
+        <CowCard
+          {...{
+            ...baseProps,
+            cow: testCow,
+            cowInventory: [testCow, breedingCow1, breedingCow2],
+            cowBreedingPen: {
+              cowId1: breedingCow1.id,
+              cowId2: breedingCow2.id,
+              daysUntilBirth: -1,
+            },
+            money: 0,
+            handleCowSellClick: () => {},
+          }}
+        />
+      )
+
+      const button = screen
+        .getByText('Breed')
+        .closest('label')
+        .querySelector('[type=checkbox]')
+      expect(button).toHaveAttribute('disabled')
+    })
+  })
 })
 
 describe('CowCardSubheader', () => {
+  let baseProps
+
   beforeEach(() => {
-    component = shallow(
-      <CowCardSubheader
-        {...{
-          cow: generateCow({
-            color: cowColors.WHITE,
-            happiness: 0,
-            name: '',
-            baseWeight: 100,
-          }),
-          cowBreedingPen: { cowId1: null, cowId2: null, daysUntilBirth: -1 },
-          cowValue: 1000,
-          isCowPurchased: false,
-        }}
-      />
-    )
+    jest.spyOn(Math, 'random').mockReturnValue(0)
+    baseProps = {
+      cow: generateCow({
+        color: cowColors.WHITE,
+        happiness: 0,
+        name: '',
+        baseWeight: 100,
+      }),
+      cowBreedingPen: { cowId1: null, cowId2: null, daysUntilBirth: -1 },
+      cowInventory: [],
+      cowValue: 1000,
+      isCowPurchased: false,
+    }
   })
 
   describe('happiness display', () => {
     describe('cow is not purchased', () => {
       test('renders no hearts', () => {
-        expect(component.find('.heart')).toHaveLength(0)
+        const { container } = render(<CowCardSubheader {...baseProps} />)
+
+        expect(container.querySelector('.heart')).toBeNull()
       })
     })
 
     describe('cow is purchased', () => {
       test('renders hearts', () => {
-        component.setProps({
-          isCowPurchased: true,
-        })
+        const { container } = render(
+          <CowCardSubheader {...{ ...baseProps, isCowPurchased: true }} />
+        )
 
-        expect(component.find('.heart')).toHaveLength(10)
+        expect(container.querySelectorAll('.heart')).toHaveLength(10)
       })
 
       test('renders full hearts that match cow happiness', () => {
-        expect(component.find('.heart.is-full')).toHaveLength(0)
+        let { container } = render(
+          <CowCardSubheader
+            {...{
+              ...baseProps,
+              cow: generateCow({
+                color: cowColors.WHITE,
+                happiness: 0.5,
+                name: '',
+                baseWeight: 100,
+              }),
+              isCowPurchased: true,
+            }}
+          />
```

Comment:
random comment, but you can do these as ```<CowCardSubheader { ...baseProps} cow={generateCow(...)}  isCowPurchased />``` instead of spreading everything. sorry if you already told me a reason some other time why you prefer the style you use 😄

Reply:
Hah this is just my preferred JSX style. :) I find it generally easier to treat all JSX properties as standard JavaScript objects. I realize that this is not the common React style, so I don't expect anyone else to use it when working on the project! 😄